### ISSUE TITLE
Medias are not playing 

### ISSUE THREAD

**Issue by artemkrachulov** [wistia/WistiaKit#36]
Hi. I need to get media by hash id, change start paying position and play video.

I use follow code to show player VC from media model but getting "white page & skull screen":

```
globalAPI.showMedia(forHash: "xxx") { media, error in
      self.wistiaPlayerVC.replaceCurrentVideoWithVideo(forMedia: media!, autoplay: true)
      self.present(self.wistiaPlayerVC, animated: true, completion: nil)
}
```
When I tested this hash with another method

```
wistiaPlayerVC.replaceCurrentVideoWithVideo(forHashedID: "xxx")
```

I noticed that received `media` models are different.

How I can do follow:
1. Get `media` by hash id
2. Set start paying position
3. Play video

**Comment by spinosa** [comment]
Hi @artemkrachulov ,

Are you using WistiaKit 0.19.1 ?  It includes an important update that has caused playback issues for others when using `WistiaPlayerViewController`

**Comment by artemkrachulov** [comment]
Updated.
Now I've got an error:

> Video Playback Problem -- Media xxx has no assets compatible with this player's configuration.

This is right because I have different api request for creating WistiaMedia models:

```
globalAPI.showMedia(forHash: "xxx") { media, error in
    . . .    
}

public : showMedia => GET https://api.wistia.com/v1/medias/xxx.json

Response: 

{
  "id":4489021,
  "name":"name",
  "type":"Video",
  "created":"2013-09-19T15:30:49+00:00",
  "updated":"2013-10-28T20:53:12+00:00",
  "duration":167.0,
  "hashed_id": "xxx",
  "description": "description",
  "progress":1.0,
  "status":"ready",
  "thumbnail": {
    "url":"http://embed.wistia.com/deliveries/7fbf9c2fe9c6585f9aa032f43f0aecc3f287e86b.jpg?image_crop_resized=100x60",
    "width":100,
    "height":60
  },
  "project":{
    "id":464427,
    "name":"How They Work",
    "hashed_id":"ln2k6qwi9k"
  },
  "assets":[{      
"url":"http://embed.wistia.com/deliveries/856970d9a4bcb9aab381a0bd9ab714f19d72c62f.bin",
      "width":960,
      "height":540,
      "fileSize":23695556,
      "contentType":"video/mp4",
      "type":"OriginalFile"
    }, . . . ]
}
```

and

```
wistiaPlayerVC.replaceCurrentVideoWithVideo(forHashedID: "xxx")

internal : mediaInfo => GET https://fast.wistia.net/embed/medias/xxx.json

{
  "media": {
    "assets": [
      {
        "type": "original",
        "slug": "original",
        "display_name": "Original file",
        "width": 1920,
        "height": 1080,
        "ext": "mov",
        "size": 608439169,
        "bitrate": 48504,
        "public": true,
        "status": 2,
        "progress": 1,
        "url": "https://embed-ssl.wistia.com/deliveries/bf1c002191c9ebda79ba9dcc2a7d75971e7e18e9.bin",
        "created_at": 1461183998
      }, . . .],
    "distilleryUrl": "https://distillery.wistia.com/x",
    "accountKey": "wistia-production_284757",
    "mediaKey": "wistia-production_19707373",
    "type": "Video",
    "progress": 1,
    "status": 2,
    "name": "VirtuePlus pitchvideo",
    "duration": 98.138,
    "hashedId": "er8rjx574f",
    "branding": false,
    "seoDescription": "a LitUpp Global video from LitUpp Development",
    "preloadPreference": null,
    "flashPlayerUrl": "https://embed-ssl.wistia.com/flash/embed_player_v2.0.swf?2015-02-27",
    "showAbout": true,
    "createdAt": 1461183998,
    "firstEmbedForAccount": false,
    "firstShareForAccount": false,
    "stats": {
      "loadCount": 332,
      "playCount": 289,
      "uniqueLoadCount": 18,
      "uniquePlayCount": 16,
      "averageEngagement": 0.0658885
    },
    "trackingTransmitInterval": 20,
    "integrations": {},
    "hls_enabled": true,
    "embed_options": {
      "volumeControl": "true",
      "fullscreenButton": "true",
      "controlsVisibleOnLoad": "true",
      "playerColor": "54bbff",
      "bpbTime": "false",
      "googleAnalytics": true,
      "vulcan": "true",
      "videoQuality": ""
    }
  },
  "options": {}
}
```

How I can play created model after `showMedia` method with `WistiaPlayerViewController`?

**Comment by spinosa** [comment]
You could use either

```swift
globalAPI.showMedia(forHash: "xxx") { media, error in
  if let m = media {
    wistiaPlayerVC.replaceCurrentVideoWithVideo(forHashedID: m.hashedID)
  }
}
```

or

```swift
globalAPI.showMedia(forHash: "xxx") { media, error in
  if let m = media {
    wistiaPlayerVC.replaceCurrentVideoWithVideo(forMedia: m)
  }
}
```

HTH!  Let me know if that works for you, @artemkrachulov

**Comment by artemkrachulov** [comment]
This code works well. I talked about this before

```
globalAPI.showMedia(forHash: "xxx") { media, error in
  if let m = media {
    wistiaPlayerVC.replaceCurrentVideoWithVideo(forHashedID: m.hashedID)
  }
}
```

But I need  change start paying position for video, therefore I need second code:

```
globalAPI.showMedia(forHash: "xxx") { media, error in
  if let m = media {
    wistiaPlayerVC.replaceCurrentVideoWithVideo(forMedia: m)
  }
}
```

This code return an error.

> Video Playback Problem -- Media xxx has no assets compatible with this player's configuration.

**Comment by spinosa** [comment]
Hmm, not sure why one approach fails to the load your media while the other doesn't.  If you can share the hashedID (email dspinosa@wistia.com if you'd prefer to keep it between us) I'll look into the discrepancy.

As for seeking, that can be done using API on the underlying `WistiaPlayer` as follows:
`wistiaPlayerVC.wPlayer.seek(to:withTolerance:completionHandler:)`
doesn't matter how you init the player, so long as it's playing, you can seek

**Comment by spinosa** [comment]
@artemkrachulov Thanks for the hashed id, helped me to identify this bug.

Version 0.20 of WistiaKit will fix the bug you're encountering.

**Comment by spinosa** [comment]
Closed by 07407c63bb98b26846b43836c3f562cfd2a424bc
